FLT/Deformations/RepresentationTheory/Irreducible.lean

```diff
@@ -0,0 +1,19 @@
+import FLT.Deformations.RepresentationTheory.Subrepresentation
+import FLT.Mathlib.RepresentationTheory.Basic
+
+namespace Representation
+
+variable {G : Type*} [Group G]
+
+variable {k : Type*} [Field k]
+
+variable {W : Type*} [AddCommMonoid W] [Module k W]
+
+class Irreducible (ρ : Representation k G W) : Prop where
+  irreducible : IsSimpleOrder (Subrepresentation ρ)
+
+class AbsolutelyIrreducible (ρ : Representation k G W) : Prop where
+  absolutelyIrreducible : ∀ k', ∀ _ : Field k', ∀ _ : Algebra k k',
+    ∀ _ : Function.Injective (algebraMap k k'), Irreducible (k' ⊗ᵣ' ρ)
```

Comment:
This definition is now weird, because if k has zero divisors then there will be no fields at all which admit an injective map from k, and thus every k-representation will be vacuously absolutely irreducible. Why not drop injectivity? It will be automatic in all the cases in which one wants to apply it, namely when k is a field, and it simplifies the definition to drop it. The class should also be called `IsAbsolutelyIrreducible` because it's a Prop, it should have a docstring, and the docstring should explain that it gives the right answer when `k` is a field and otherwise should be used at your own risk.

Reply:
As of right now, the definition needs k to be a field right? So I don't understand the "and otherwise should be used at your own risk." part

Reply:
I do see how injectivity is now free, happy to drop that

Reply:
Extra question: How could k have zero divisors if k is a field?

Reply:
Oh apologies! I had missed that `k` was already a field, so some of what I said above is nonsense (I'd just read through the subrepresentation file, where k was a commring). Sorry for creating confusion. My comment should have just said "drop injectivity, this is automatic for fields". The definition is fine.